### ISSUE TITLE
Add flag to ignore 'Skipping analyzing ...'

### ISSUE THREAD

**Issue by mfarrugi** [python/mypy#9789]
**Pitch**

Currently `--skip-missing-imports` silences all 3 different errors listed in [missing-imports](https://mypy.readthedocs.io/en/stable/running_mypy.html?highlight=skipping%20analyzing#missing-imports), but [missing-type-hints-for-third-party-library](https://mypy.readthedocs.io/en/stable/running_mypy.html?highlight=skipping%20analyzing#missing-type-hints-for-third-party-library) is closer to a warning than an error.

The current solution is to list the packages to ignore in a configuration file, which is particularly onerous when there are many packages being built and type-checked.

**Feature**

Add a flag (eg. `--skip-untyped-imports`) that either ignores this error, or downgrades it to a warning:
```
error: Skipping analyzing X: found module but no type hints or library stubs
```

Alternatively:
- the [unable-to-find-module](https://mypy.readthedocs.io/en/stable/running_mypy.html?highlight=skipping%20analyzing#unable-to-find-module) error could be changed to no longer be skipped by `--skip-missing-imports`, as that is nearly always a misconfiguration of mypy
- each case could be guarded by a separate flag. 

**Implementation**

https://github.com/python/mypy/blob/master/mypy/build.py#L2410-L2411 and other references to `options.ignore_missing_imports` would need to be changed.

**Comment by hauntsaninja** [comment]
Reasonable request, although I'm wary of adding more knobs to mypy's import following... I think in practice people don't even understand the existing ones very well and they're a common source of confusion. I'd also phrase it as `--ignore-untyped-imports`.
Note that of those three errors, you won't run into one of them since typeshed is virtually complete wrt stdlib right now. In fact, I recently removed documentation of that error in #9675.
Finally, I'd personally contend with the assertion that it's closer to a warning than an error :-) All three of them have the same potential effect: causing a lot of your code to go unchecked.

**Comment by mfarrugi** [comment]
I think the distinction I see is that "untyped-imports" is not _really_
something most users can/will fix, but "missing-imports" almost certainly
means something is broken that the user needs to fix. I would advocate for
removing `--ignore-missing-imports`, but the docs vaguely allude to errors
in import finding that can't be fixed otherwise.

**Comment by Ajk4** [comment]
+1

I don't have control over 3rd party libraries. But I still want to use mypy to check code I have control over. Even if 3rd party library doesn't have type hints.

Use case -> I want to check code I have control over in CI.

**Comment by cpoptic** [comment]
I'd counter that this is an excellent feature request.  Here's why:

Often what mypy calls "errors" of "missing stubs" involve a 3rd party external library that the user is merely importing to extend their own code.

It boils down to how much the user trusts these 3rd party libraries.

For instance, I've gotten the `Skipping analyzing "numpy": module is installed, but missing library stubs` error when importing the `numpy` and `pandas` libraries.  Those are some of most heavily used python libraries and most users ascribe to them a reasonable amount of trust that they work.  Therefore, mypy should not throw errors for those library imports.

Now I will concede that other 3rd party libraries that are not as well tested as the two above could very well cause a lot of your code to go unchecked, and therefore it would be reasonable for mypy to throw an error if those libraries are missing stubs.

So in summary, `numpy` and `pandas` should probably not throw errors if missing library stubs.
But `my_random_untested_3rd_party_library` probably should, at least by default.

Also I agree with @mfarrugi, as most users will probably not bother fixing "untyped-imports" from a 3rd party library that they are likely not familiar with aside from how to call functions to use within their code.
But "missing-imports" is obviously a much more serious error that they will fix.
The two errors do not have the same level of "actionable" fixes.

**Comment by hauntsaninja** [comment]
On mypy master, you can use `--disable-error-code=import-untyped` to silence all errors for third party packages that are located but missing stubs.
